backstory/src/main/java/com/google/sps/story/data/DatamuseRequestClient.java

```diff
@@ -55,61 +59,79 @@ public DatamuseRequestClient(String url) {
   }
 
   /**
-   * Fetches an array of adjectives related to the passed-in noun, which will have
-   * a passed-in cap on its size (array returned might be smaller if database
-   * does not have enough related adjectives). Fetches these adjectives
-   * by querying the Datamuse database.
+   * Returns a randomly-chosen topic related to storytelling
+   * from a preset array.
    *
-   * @param noun the noun to get adjectives related to (must be one word)
-   * @param cap the maximum number of adjectives to retrieve
-   * @return an array of adjectives related to the noun
+   * @return a topic related to storytelling to be used in the PCS
+   */
+  public static String getStorytellingTopic() {
+    int randomIndex = (int) (Math.random() * STORYTELLING_TOPICS.length);
+
+    return STORYTELLING_TOPICS[randomIndex];
+  }
+
+  /**
+   * Fetches an array of words, of type wordType, related to the passed-in noun,
+   * which will have a passed-in cap on its size (array returned might be smaller if database
+   * does not have enough related adjectives). Fetches of these the words most related
+   * to the passed-in topic. Accomplished by querying the Datamuse database.
+   *
+   * @param noun the noun to get words related to (must be one word)
+   * @param wordType the type of word to fetch (e.g. adjectives or gerunds)
+   * @param cap the maximum number of words to retrieve
+   * @param topic will sort the words most relevant to given topic (so top ten will
+   *    be words related to the noun then the ones most related to the given topic)
+   *    (empty string for topic is same as no topic)
+   * @return an array of words of type wordType related to the noun of max size cap
    * @throws IllegalArgumentException if noun is more than one word (has whitespace)
    * @throws APINotAvailableException if Datamuse API cannot be reached
    * @throws RuntimeException if JSON received back from the API cannot be parsed (JSONException)
    *    or does not have objects of type JSON (ClassCastException)
    */
-  public String[] fetchRelatedAdjectives(String noun, int cap)
-      throws IllegalArgumentException, APINotAvailableException, RuntimeException {
-    // check for whitespace
-    Pattern pattern = Pattern.compile("\\s");
-    Matcher matcher = pattern.matcher(noun);
-
-    if (matcher.find()) {
-      throw new IllegalArgumentException("Noun cannot contain whitespace (must be one word).");
+  public String[] fetchRelatedWords(String noun, DatamuseRequestWordType wordType, int cap,
+      String topic) throws IllegalArgumentException, APINotAvailableException, RuntimeException {
+    validateWordArgument(noun, "Noun");
+
+    String query = url;
+
+    switch (wordType) {
+      case ADJECTIVE:
+        query += "rel_jjb=" + noun; // get adjectives related to the noun
+        break;
+      case GERUND:
+        query += "rel_jja=" + noun + "&sp=*ing"; // get nouns related to the noun ending in "ing"
+        break;
     }
```

Comment:
Adding a default throwing case would help protect against developers forgetting to update this switch when adding new enum values:

```
default:
  throw new IllegalArgumentException(...);
```

Reply:
Great idea! I'll add this. Thanks!

Reply:
Done

Reply:
Done